biggerquery/interactive.py

```diff
@@ -0,0 +1,785 @@
+
+        return component_return_value
+
+    component_callable_with_proper_signature = "def reworked_function({signature}):\n    return original_func({kwargs})\n".format(
+        signature=','.join([arg for arg in getargspec(standard_component).args]),
+        kwargs=','.join(['{arg}={arg}'.format(arg=arg) for arg in getargspec(standard_component).args]))
+    component_callable_with_proper_signature_code = compile(component_callable_with_proper_signature, "fakesource",
+                                                            "exec")
+    fake_globals = {}
+    eval(component_callable_with_proper_signature_code, {'original_func': component_callable}, fake_globals)
+
+    component_callable = fake_globals['reworked_function']
+    component_callable.__name__ = operation_name or standard_component.__name__
+
+    return results_container, component_callable
+
+
+class OperationLevelDatasetManager(object):
+    """
+    Let's you run specified operation or peek a result of a specified operation.
+    """
+
+    def __init__(self, dataset_manager, peek=False, operation_name=None, peek_limit=DEFAULT_PEEK_LIMIT):
+        self._dataset_manager = dataset_manager
+        self._peek = peek
+        self._operation_name = operation_name
+        self._peek_limit = peek_limit
+        self._results_container = []
+
+    def write_truncate(self, table_name, sql, partitioned=True, custom_run_datetime=None, operation_name=None):
+        return self._run_write_operation(
+            operation_name=operation_name,
+            method=self._dataset_manager.write_truncate,
+            sql=sql,
+            table_name=table_name,
+            partitioned=partitioned,
+            custom_run_datetime=custom_run_datetime)
+
+    def write_append(self, table_name, sql, partitioned=True, custom_run_datetime=None, operation_name=None):
+        return self._run_write_operation(
+            operation_name=operation_name,
+            method=self._dataset_manager.write_append,
+            sql=sql,
+            table_name=table_name,
+            partitioned=partitioned,
+            custom_run_datetime=custom_run_datetime)
+
+    def write_tmp(self, table_name, sql, custom_run_datetime=None, operation_name=None):
+        return self._run_write_operation(
+            operation_name=operation_name,
+            method=self._dataset_manager.write_tmp,
+            sql=sql,
+            table_name=table_name,
+            custom_run_datetime=custom_run_datetime)
+
+    def collect(self, sql, custom_run_datetime=None, operation_name=None):
+        return self._run_write_operation(
+            operation_name=operation_name,
+            method=self._dataset_manager.collect,
+            sql=sql,
+            custom_run_datetime=custom_run_datetime)
+
+    def create_table(self, create_query, operation_name=None):
+        if self._operation_name == operation_name or self._operation_name is None:
```

Comment:
`_should_run_operation` method should be used

Reply:
ok

Reply:
done